ENG-966 / ENG-967 - AI Hackstack create mode in teacher dashboard

adamkecskes opened the pull request:
- New icon for 'use' mode scenarios.
 - Show them completed if `isReadyToReview` is checked.

AI PR: https://github.com/codecombat/ai/pull/264
Backend PR: https://github.com/codecombat/codecombat-server/pull/1020
<img width="932" alt="Notification_Center" src="https://github.com/user-attachments/assets/c870bf11-4248-4324-945d-b12189e29b89">


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit


- **New Features**
  - Added localization support with new phrases to enhance user engagement.
  - Enhanced project status display with contextual information based on the current scenario mode in the teacher dashboard.
  
- **Improvements**
  - Improved conditional rendering in components for a more dynamic user interface.
  - Refined logic for adjusting learning goals based on project scenarios.

- **Bug Fixes**
  - Resolved issues related to icon rendering conditions in the content display component.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mrfinch commented on ozaria/site/components/teacher-dashboard/Panel/components/AiProject.vue:
```diff
@@ -1,14 +1,23 @@
-      {{ $t('teacher_dashboard.no_failed_attempts') }}
+
+    <p v-if="mode === 'use'">
+      {{ aiProject.isReadyToReview ? $t('teacher_dashboard.ready_to_review') : $t('teacher.in_progress') }}
```
seems like u missed adding i18n for this one in en.js

mrfinch commented on ozaria/site/components/teacher-dashboard/Panel/components/AiProject.vue:
```diff
@@ -1,14 +1,23 @@
     </p>
+    <div v-else>
+      <p>
+        Progress: {{ progress }}%
```
can we make this p bold to stand out in this section?

mrfinch reviewed (commented):
- we should remove 'Learning Goal' section if it's empty
- why does use-gpt4 come twice? we should show it only once?

<img width="373" alt="Screenshot 2024-08-14 at 10 20 58 AM" src="https://github.com/user-attachments/assets/7a1fb6f6-513c-426c-8312-f3a03da3d86e">

adamkecskes commented:
@mrfinch 
> why does use-gpt4 come twice? we should show it only once?
<img width="510" alt="ENG-966___ENG-967_-_AI_Hackstack_create_mode_in_teacher_dashboard_by_adamkecskes_·_Pull_Request__7630_·_codecombat_codecombat" src="https://github.com/user-attachments/assets/90151995-01c2-426b-816e-cd819451f555">
> we should remove 'Learning Goal' section if it's empty
Removed for "Create" mode, but I've added `aiScenario.name` for "Learn" mode projects:
<img width="562" alt="CodeCombat_Teacher_Dashboard" src="https://github.com/user-attachments/assets/e041d23b-0ea5-4252-abd6-b5ce89a34043">
